pkg/controller/controller_utils.go

```diff
@@ -1112,6 +1113,29 @@ func AddPodControllerUIDIndexer(podInformer cache.SharedIndexInformer) error {
 	})
 }
 
+// FilterPodsByOwner gets the Pods managed by an owner or orphan Pods in the owner's namespace
+func FilterPodsByOwner(podIndexer cache.Indexer, owner *metav1.ObjectMeta) ([]*v1.Pod, error) {
```

Comment:
Can we take this refactoring as an opportunity to introduce a simple unit test?

Reply:
Added in the last commit https://github.com/kubernetes/kubernetes/pull/132615/commits/e8e178e410018a55201d913afcc5e40ca11dffb4, ptal

Reply:
Thanks!